workforceDashboard/src/main/java/gov/homeoffice/workforceDashboard/service/EmployeeServiceImpl.java

```diff
@@ -24,4 +30,112 @@ public void saveEmployee (Employee employee){
+
+            //Create a new employee object
+            Employee employee = new Employee();
+
+            //For each row, iterate through all the columns
+            for (int i = 1; i <= sheet.getLastRowNum(); i++) {
+                row = sheet.getRow(i);
+
+                String roleRef = row.getCell(0).getStringCellValue();
+                String roleTitle = row.getCell(1).getStringCellValue();
+                String employeeFirstName = row.getCell(2).getStringCellValue();
+                String employeeSurname = row.getCell(3).getStringCellValue();
+                String employeeFullName = row.getCell(4).getStringCellValue();
+                Double employeeAdelphiNumber = row.getCell(5).getNumericCellValue();
+                String employeeEmail = row.getCell(6).getStringCellValue();
+                String gradeEquivalent = row.getCell(7).getStringCellValue();
+                String function = row.getCell(8).getStringCellValue();
+                String businessUnit = row.getCell(9).getStringCellValue();
+                String team = row.getCell(10).getStringCellValue();
+                String professionCluster = row.getCell(11).getStringCellValue();
+                String dDaTProfessionRole = row.getCell(12).getStringCellValue();
+                String affiliatedDdaTProfessionRole = row.getCell(13).getStringCellValue();
+                String currentResourceRollUp = row.getCell(14).getStringCellValue();
+                String functionRollUp = row.getCell(15).getStringCellValue();
+                String employeeCurrentPrimaryLocation = row.getCell(16).getStringCellValue();
+                String employeeAnticipatedFutureLocation = row.getCell(17).getStringCellValue();
+                String eUExit = row.getCell(18).getStringCellValue();
+                String isThisRoleAVacancy = row.getCell(19).getStringCellValue();
+                String vacancyType = row.getCell(20).getStringCellValue();
+                Double vacancyStage = row.getCell(21).getNumericCellValue();
+                String rechargedRoles = row.getCell(22).getStringCellValue();
+                String operationalLineManagerFirstName = row.getCell(23).getStringCellValue();
+                String operationalLineManagerSurname = row.getCell(24).getStringCellValue();
+                String operationalLineManagerRoleReference = row.getCell(25).getStringCellValue();
+                String currentResourceType = row.getCell(26).getStringCellValue();
+                String targetResourceType = row.getCell(27).getStringCellValue();
+                String projectedStartDateOfRole = row.getCell(28).getStringCellValue();
+                String projectedEndDateOfTerminatingRole = row.getCell(29).getStringCellValue();
+                Double fTE = row.getCell(30).getNumericCellValue();
+                String dDaTOrNonDDaTResource = row.getCell(31).getStringCellValue();
+
+                employee.setRoleRef(roleRef);
```

Comment:
We can half the number of lines here by assigning the cell value directly to the setter method. Saves us from declaring all of them first and referring to the same value twice.

Reply:
Completed see latest commit